### PR TITLE
Dual and Many module variants added, all settings.gradle files reading added.

### PR DESCRIPTION
**Pull request by Malligan**

### CODE REVIEW

File: Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt [hunk 1]
```diff
@@ -1,59 +1,70 @@
 package com.alged.bfe
 
 import com.intellij.lang.ASTNode
+import com.intellij.psi.PsiFile
 
-class ModulesSelectionService {
+class ModulesSelectionService(var configuration: ModulesSelectionConfiguration) {
```
Comment (egorikftp) [inline Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt]: can be 
```suggestion
class ModulesSelectionService(private val configuration: ModulesSelectionConfiguration) {
```

File: Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt [hunk 2]
```diff
@@ -1,59 +1,70 @@
...
-
-    var tableModel : MutableList<RowModelWithNode> = mutableListOf()
+    fun buildTableModel(file: PsiFile): MutableList<RowModelWithNode> {
+        val localTableModel: MutableList<RowModelWithNode> = mutableListOf()
+        val fileASTNode = file.node
+        val nodes = fileASTNode.getChildren(null)
 
-    fun updateTableModel(nodes : Array<ASTNode>): List<Array<Any>> {
-        val projectNodes : MutableList<ASTNode>  = (nodes.filter { it.text.contains("project") }).toMutableList()
+        val projectNodes: MutableList<ASTNode>  = (nodes.filter { it.text.contains("project") }).toMutableList()
```
Comment (egorikftp) [inline Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt]:
```suggestion
        val projectNodes: MutableList<ASTNode> = nodes.filter {
            it.text.contains("project")
        }.toMutableList()
```

File: Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt [hunk 3]
```diff
@@ -1,59 +1,70 @@
...
+        for (item in itemsForModel) {
+            tableModel.add(RowModelWithNode(item.second, !item.first.text.contains("//"), item.first, item.third))
+            nodes.remove(item.first)
+        }
     }
 }
 
-data class RowModelWithNode(val moduleName: String, val moduleEnabled: Boolean, val moduleNode: ASTNode)
\ No newline at end of file
+data class RowModelWithNode(val moduleName: String,
```
Comment (egorikftp) [inline Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt]: Can we extract this to separate `model` package?)

File: Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt (hunk 1 above)
Comment (Malligan) [reply Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt]: In my first ideas configuration was updatable. Seems like we can create new Service.

File: Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt (hunk 3 above)
Comment (Malligan) [reply Build file explorer/src/com/alged/bfe/ModulesSelectionService.kt]: Seems like we can.

**Pull request merged**